### PR TITLE
Remove old "ApiEvent" proto

### PR DESCRIPTION
**Pull request by florian-kuebler**
This proto was used for the old manual instrumentation implementation. Newer implementations of the old version of manual instrumentation are not producing these events anymore. So taking new captures with an old instrumentated target is completely fine.

This change breaks compatability with very old captures. However, those captures must be more than one year old (see #2662).

Test: Compile & run tests

### CODE REVIEW

File: src/ApiUtils/include/ApiUtils/EncodedEvent.h [hunk 1]
```diff
@@ -13,82 +13,6 @@
...
-  orbit_api::EventType Type() const { return encoded_event.Type(); }
-
-  EncodedEvent encoded_event;
-  uint32_t pid;
-  uint32_t tid;
-  uint64_t timestamp_ns;
-};
-
 template <typename Dest, typename Source>
 inline Dest Encode(const Source& source) {
```
Comment (dpallotti) [inline src/ApiUtils/include/ApiUtils/EncodedEvent.h:17]: `Encode` and `Decode` are now only used in `CaptureEventProcessorForListener` and in `OrbitGl` for the memory events (which I remember disagreeing with...), right? I think the remaining of this file should be moved, maybe in `CaptureClient` if the dependencies allow it? If this is something you don't feel like tackling, a big TODO and explanation of the fact that they are out of place seems appropriate. (Of course the cleaner solution would be to get rid of them and rework the representation of the memory events, but that's a different story.

**Review by dpallotti — APPROVED** [review]
Nice, see comment

File: src/ApiUtils/include/ApiUtils/EncodedEvent.h (hunk 1 above)
Comment (florian-kuebler) [reply src/ApiUtils/include/ApiUtils/EncodedEvent.h:17]: I will actually remove the Encode/Decode completely in  one of my next PRs, as this is a bit of a bigger thing and kinda unrelated to this here, I'd keep this PR as it is.

Should I still add a TODO?

Comment (dpallotti) [reply src/ApiUtils/include/ApiUtils/EncodedEvent.h:17]: If you are already planning to tackle this, a short TODO or no TODO are fine.

Comment (florian-kuebler) [reply src/ApiUtils/include/ApiUtils/EncodedEvent.h:19]: Done.

**Pull request merged**